urllib3/fields.py

```diff
@@ -36,11 +36,11 @@ def format_header_param(name, value):
         result = '%s="%s"' % (name, value)
         try:
             result.encode('ascii')
-        except UnicodeEncodeError:
+        except (UnicodeEncodeError, UnicodeDecodeError):
             pass
         else:
             return result
-    if not six.PY3:  # Python 2:
+    if not six.PY3 and isinstance(value, unicode):  # Python 2:
```

Comment:
I took a look at the logs; looks like it's failing because unicode isn't a valid type in Py3. We could definitely get around THAT by switching to `six.text_type`, but yeah, we'd still need `not six.PY3`.